packages/mui-system/src/index.d.ts

```diff
@@ -41,11 +43,11 @@ export function breakpoints<Props, Breakpoints extends string = DefaultBreakPoin
   styleFunction: StyleFunction<Props>,
 ): StyleFunction<Partial<Record<Breakpoints, Props>> & Props>;
 
-// restructures the breakpoints in the in the correct order and merges all styles args
+// restructures the breakpoints in the correct order and merges all styles args
 export function mergeBreakpointsInOrder(
-  breakpointsInput: { keys: string[]; up: (key: string) => string },
-  ...styles: object[]
-): object;
+  breakpointsInputs: Breakpoints,
+  styles: CSSObject[],
+): CSSObject;
```

Comment:
Let's not duplicate the type definitions. We can re-export the definition from `breakpoints` file:
```diff
--- a/packages/mui-system/src/index.d.ts
+++ b/packages/mui-system/src/index.d.ts
@@ -32,7 +32,7 @@ export type BordersProps = PropsFor<typeof borders>;

 // breakpoints.js
 type DefaultBreakPoints = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
-export { handleBreakpoints } from './breakpoints';
+export { handleBreakpoints, mergeBreakpointsInOrder } from './breakpoints';

 /**
  * @returns An enhanced stylefunction that considers breakpoints
@@ -41,12 +41,6 @@ export function breakpoints<Props, Breakpoints extends string = DefaultBreakPoin
   styleFunction: StyleFunction<Props>,
 ): StyleFunction<Partial<Record<Breakpoints, Props>> & Props>;

-// restructures the breakpoints in the in the correct order and merges all styles args
-export function mergeBreakpointsInOrder(
-  breakpointsInput: { keys: string[]; up: (key: string) => string },
-  ...styles: object[]
-): object;
-
 export function compose<T extends Array<StyleFunction<any>>>(...args: T): ComposedStyleFunction<T>;

 export type DisplayProps = PropsFor<typeof display>;
```